hangman.py

```diff
@@ -0,0 +1,222 @@
+    
+    Starts up an interactive game of Hangman.
+    
+    * At the start of the game, let the user know how many 
+      letters the secret_word contains and how many guesses s/he starts with.
+      
+    * The user should start with 6 guesses
+
+    * Before each round, you should display to the user how many guesses
+      s/he has left and the letters that the user has not yet guessed.
+    
+    * Ask the user to supply one guess per round. Remember to make
+      sure that the user puts in a letter!
+    
+    * The user should receive feedback immediately after each guess 
+      about whether their guess appears in the computer's word.
+
+    * After each guess, you should display to the user the 
+      partially guessed word so far.
+    
+    Follows the other limitations detailed in the problem write-up.
+    '''
+    warnings = 3
+    attemps = 6
+    letters_guessed = set()
+    print(f'I am thinking of a word that is {len(secret_word)} letters long')
+    print(f'Your have {warnings} warnings')
+    while attemps > 0 and not is_word_guessed(secret_word, letters_guessed):
+        print("-" * 13)
+        print("You have ", attemps, " guesses left.")
+        print("Available letters: ", get_available_letters(letters_guessed))
+        letter = input("enter letter:").strip(' ')
+        letter.lower()
+        if hints and letter == HINTS:
+            show_possible_matches(get_guessed_word(secret_word, letters_guessed).replace(" ", ""))
+        elif not letter.isalpha() or letter in letters_guessed:
+            if warnings > 0:
+                warnings -= 1
+                print("Oops! That is not a valid letter. You have", warnings, "warnings left")
+            else:
+                attemps -= 1
+        else:
+            letters_guessed.add(letter)
+            splited_word = get_guessed_word(secret_word, letters_guessed)
+            print(splited_word)
+            guessed = splited_word
+            if letter not in secret_word:
+                if letter in VOWELS:
+                    attemps -= 1
+                attemps -= 1
+    if attemps <= 0:
+        print(f'Sorry, you ran out of guesses. The word was {secret_word}')
+    else:
+        points = int(attemps) * len(set(secret_word))
+        print("Congratulations, you won! Your total score for this game is:", points)
+
+
+def match_with_gaps(my_word, other_word):
+    '''
+    my_word: string with _ characters, current guess of secret word
+    other_word: string, regular English word
+    returns: boolean, True if all the actual letters of my_word match the 
+        corresponding letters of other_word, or the letter is the special symbol
+        _ , and my_word and other_word are of the same length;
+        False otherwise: 
+    '''
+    if len(my_word) == len(other_word):
+        my_word_set = set(my_word)
+        for i in range(len(my_word)):
+            if my_word[i] == LOWOER_SPACE:
+                if other_word[i] in my_word_set:
```

Comment:
This kind of functions are really hard to read. It would be better to add inline comments to our code.